File: ECE39500/exams/bak_midterm-2/midterm-2/PartA/CVCVState.py
from CompilerState import CompilerState
from StateFactory import StateFactory
import sys

class CVCVState(CompilerState):
    INT_MIN = -sys.maxsize - 1

    def __init__(self, compiler):
        self.compiler = compiler
        return
    
    def comma(self):
        self.compiler.state = StateFactory.getState("VCVState")

    def value(self, s):
        print("Parsing CVCV, expected , found int {s}")
        self.compiler.state = StateFactory.getState("ErrorState")

    def eol(self):
        print(f"Parsing CVCV, expected value, found eol")
        self.compiler.state = StateFactory.getState("ErrorState")

    def eof(self):
        print("Parsing InitState, expected value, found eof")
        self.compiler.state = StateFactory.getState("ErrorState")
    
    def __str__(self):
        return "CVCVState"

File: ECE39500/exams/midterm-2/PartB/src/CVCVState.java
public class CVCVState extends CompilerState {

   Compiler compiler = null; 

   public CVCVState(Compiler compiler) {
      this.compiler = compiler;
   }

   public void comma( ) { 
      compiler.state = StateFactory.getState("VCVState");
   }

   public void value(String str) {
      System.out.println("Parsing CVCV, expected , found int "+str);
      compiler.state = StateFactory.getState("ErrorState");
   }

   public void eol( ) { 
      System.out.println("Parsing CVCV, expected value, found eol");
      // XXX compiler.targetState.eol( );
      compiler.state = StateFactory.getState("ErrorState");
   }
      
   public void eof( ) {
      System.out.println("Parsing CVCV, expecting value, found eof");
      compiler.state = StateFactory.getState("ErrorState");
   }

   public String toString( ) {
      return "CVCVState";
   }
}
